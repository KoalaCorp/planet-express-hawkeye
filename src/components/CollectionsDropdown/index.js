import React, { useState, useEffect } from "react"
import PropTypes from "prop-types"

import Collections from "../Collections"
import DropdownButton from "../DropdownButton"
import DropdownError from "../DropdownError"

import useFetchCollections from "../../hooks/fetch-collections"

const CollectionsDropdown = ({ collectionsSearchParam }) => {
  const { collections, errorsFetchCollections } = useFetchCollections()
  const [selectedCollections, setSelectedCollections] = useState([])
  const [dropdownPrompt, setDropdownPrompt] = useState("all medium selected")

  useEffect(() => {
    const message =
      selectedCollections.length === collections.length
        ? "all medium selected"
        : selectedCollections.length > 0
        ? `${selectedCollections.length} medium selected`
        : "No medium selected"
    setDropdownPrompt(message)
  }, [selectedCollections, collections])

  useEffect(() => {
    setSelectedCollections(collectionsSearchParam)
  }, [collectionsSearchParam])

  const handleCollectionInputChange = (e) => {
    if (e.target.checked) {
      setSelectedCollections([...selectedCollections, e.target.value])
    } else {
      setSelectedCollections(
        selectedCollections.filter(
          (collection) => collection !== e.target.value
        )
      )
    }
  }

  return (
    <div className="collectionsDropdown dropdown__menu--hidden">
      <DropdownButton>{dropdownPrompt}</DropdownButton>

      <div className="dropdown__menu">
        <DropdownError visible={!selectedCollections.length}>
          You have to select at least one medium to search
        </DropdownError>

        <Collections
          collections={collections}
          selectedCollections={selectedCollections}
          handleCollectionInputChange={handleCollectionInputChange}
          errors={errorsFetchCollections}
        />
      </div>
    </div>
  )
}

export default CollectionsDropdown

CollectionsDropdown.defaultProps = {
  collectionsSearchParam: [],
}

CollectionsDropdown.propTypes = {
  collectionsSearchParam: PropTypes.array,
}
