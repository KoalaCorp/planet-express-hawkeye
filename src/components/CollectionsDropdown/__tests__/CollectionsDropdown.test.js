import React from "react"
import { render } from "../../../../tests/helpers"

import CollectionsDropdown from "../"

const renderComponent = ({ ...props } = {}) =>
  render(<CollectionsDropdown {...props} />)

describe("CollectionsDropdown", () => {
  it("renders correctly", () => {
    const dropdown = renderComponent()
    expect(dropdown).toMatchSnapshot()
  })
})
