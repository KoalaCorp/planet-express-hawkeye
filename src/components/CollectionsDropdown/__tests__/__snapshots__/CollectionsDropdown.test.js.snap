// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsDropdown renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="collectionsDropdown dropdown__menu--hidden"
      >
        <button
          class="dropdown__button"
        >
          all medium selected
        </button>
        <div
          class="dropdown__menu"
        >
          <div
            class="dropdown__error dropdown__error--visible"
          >
            You have to select at least one medium to search
          </div>
          <div
            class="collections"
          >
            <div
              class="colletions__errors"
            >
              TypeError: Cannot read property 'planetExpressCharts' of undefined
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="collectionsDropdown dropdown__menu--hidden"
    >
      <button
        class="dropdown__button"
      >
        all medium selected
      </button>
      <div
        class="dropdown__menu"
      >
        <div
          class="dropdown__error dropdown__error--visible"
        >
          You have to select at least one medium to search
        </div>
        <div
          class="collections"
        >
          <div
            class="colletions__errors"
          >
            TypeError: Cannot read property 'planetExpressCharts' of undefined
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
